### PR TITLE
Implement getter function for external string user pointer

### PR DESCRIPTION
**Pull request by zherczeg**

### CODE REVIEW

File: docs/02.API-REFERENCE.md [hunk 1]
```diff
@@ -4283,6 +4283,77 @@ main (void)
...
+  - In some cases (e.g. when the string is also a magic string registered by
+    [jerry_register_magic_strings](#jerry_register_magic_strings)), the
+    string is a normal string without a user pointer even if it is created
+    by [jerry_create_external_string](#jerry_create_external_string).
+
+**Prototype**
+
+```c
+void *jerry_string_get_external_user_pointer (const jerry_value_t value,
+                                              bool *is_external);
```
Comment (galpeter) [inline docs/02.API-REFERENCE.md:4307]: Do we even need the `is_external` at all? As far as I understand this is only `true` if the return value is not-null.

Comment (zherczeg) [reply docs/02.API-REFERENCE.md:4307]: NULL is a valid value for `user_p`, so that is not enough.

Comment (galpeter) [reply docs/02.API-REFERENCE.md:4307]: And if it is a NULL value do we even care?

**Review by rerobika — APPROVED** [review]

**Review by galpeter — APPROVED** [review]
lgtm

**Pull request merged**